storage/options.go

```diff
@@ -555,6 +582,7 @@ func (o *options) MultiReaderIteratorPool() encoding.MultiReaderIteratorPool {
 
 func (o *options) SetIdentifierPool(value ident.Pool) Options {
 	opts := *o
+	opts.indexOpts = opts.indexOpts.SetIdentifierPool(value)
```

Comment:
This is non-standard, right? Any reason you're updating both and not just expecting IndexOptions with a correctly configured identifier pool?

Reply:
we do stuff like this in a couple of places in this file - e.g. `SetClockOptions`: 

```
func (o *options) SetClockOptions(value clock.Options) Options {
	opts := *o
	opts.clockOpts = value
	opts.commitLogOpts = opts.commitLogOpts.SetClockOptions(value)
	opts.seriesOpts = NewSeriesOptionsFromOptions(&opts, nil)
	return &opts
}
```

Reply:
Yeah fair enough, especially after our debugging session today